client/common/src/types.rs

```diff
@@ -1,4 +1,192 @@
-use trie_db::{Hasher, TrieLayout};
+use std::fmt::Debug;
+
+use codec::decode_from_bytes;
+use codec::{Decode, Encode};
+use serde::{Deserialize, Serialize};
+use sp_core::Hasher;
+use sp_core::H256;
+use sp_trie::CompactProof;
+use trie_db::TrieLayout;
+
+/// The file chunk size in bytes. This is the size of the leaf nodes in the Merkle tree.
+pub const FILE_CHUNK_SIZE: usize = 1024 * 1024;
 
 /// The hash type of trie node keys
-pub type HashT<T> = <<T as TrieLayout>::Hash as Hasher>::Out;
+pub type HashT<T> = <T as TrieLayout>::Hash;
+pub type HasherOutT<T> = <<T as TrieLayout>::Hash as Hasher>::Out;
+
+// TODO: this is currently a placeholder in order to define Storage interface.
+/// FileKey is the identifier for a file.
+/// Computed as the hash of the FileMetadata.
+pub type Key = Vec<u8>;
+
+// TODO: this is currently a placeholder in order to define Storage interface.
+/// This type mirrors the `FileLocation<T>` type from the runtime, which is a BoundedVec.
+type FileLocation = Vec<u8>;
+
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, Default)]
+pub struct Fingerprint(H256);
```

Comment:
Fingerprint is a wrapper over the `H256` type, which is essentially a wrapper over `[u8; 32]`. Why not making `Fingerprint = Vec<u8>` or `Fingerprint = [u8; 32]`?

Reply:
After some discussion with @links234 we decided to do it this way to ensure that only `Fingerprint` are compared to others of the same type. Meaning you wouldn't be able to compare a root (e.g. `&[u8]` or `H::Out`) with a `Fingerprint` unless you call `into()` from a slice, making it less likely to make an unintentional comparison.

And putting `H256` here conveys more meaning since it is a hash not just some raw bytes that could be represent anything.

Reply:
Conclusion: let's make `Fingerprint([u8; 32])`